bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1695,28 +1695,22 @@ std::string ReplicaImp::getReplicaState() const {
 }
 
 void ReplicaImp::onInternalMsg(GetStatus &status) const {
-  if (status.key == "replica") {  // TODO: change this key name (coordinate with deployment)
-    return status.output.set_value(getReplicaLastStableSeqNum());
-  }
-
-  if (status.key == "replica-state") {
-    return status.output.set_value(getReplicaState());
-  }
-
-  if (status.key == "state-transfer") {
-    return status.output.set_value(stateTransfer->getStatus());
-  }
-
-  if (status.key == "key-exchange") {
-    return status.output.set_value(KeyExchangeManager::instance().getStatus());
-  }
-
-  if (status.key == "pre-execution") {
-    return status.output.set_value(replStatusHandlers_.preExecutionStatus(getAggregator()));
-  }
-
+  // TODO: make sure that no one relies on the functions being called lazily,
+  // otherwise use lambdas
+  static const std::unordered_map<const char *, std::string> status_to_output{
+      {"replica", getReplicaLastStableSeqNum()},
+      {"replica-state", getReplicaState()},
+      {"state-transfer", stateTransfer->getStatus()},
+      {"key-exchange", KeyExchangeManager::instance().getStatus()},
+      {"pre-execution", replStatusHandlers_.preExecutionStatus(getAggregator())},
+  };
   // We must always return something to unblock the future.
-  return status.output.set_value("** - Invalid Key - **");
+  static const std::string invalid_key_output = "** - Invalid Key - **";
+
+  auto *status_key = status.key.c_str();
+  auto &output = status_to_output.find(status_key) != status_to_output.end() ? status_to_output.at(status_key)
+                                                                             : invalid_key_output;
+  return status.output.set_value(output);
```

Comment:
This seems unrelated form the purpose of this PR. If so, please remove and publish as part of a separate PR.